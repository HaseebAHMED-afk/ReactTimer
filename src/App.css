.App {
  text-align: center;
  padding: 150px;
}
.main-heading{
  font-family: Verdana, Geneva, Tahoma, sans-serif;
  color: rgb(58, 58, 57);
}