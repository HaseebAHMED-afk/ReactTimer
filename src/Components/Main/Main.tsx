import React, { useState, useEffect } from 'react'
import CalculateTimer from '../Tools/CalculateTimer';
import Controls from '../Controls/Control';
import '../../App.css'

const Main: React.FC = (): JSX.Element => {
      const [timeInSeconds, setTimeInSeconds] = useState<number>(0);
      const [timerArray, setTimerArray] = useState<Array<number|string>>([]);

      useEffect(() => {
           let timeArray: Array<number|string> = CalculateTimer(timeInSeconds);
           setTimerArray(timeArray);
      },[timeInSeconds])

      return (
            <div>
            <div className="time-container">
                  <h1 className="timer">{timerArray[0]}:{timerArray[1]}:{timerArray[2]}</h1>
            </div>
            <Controls time={setTimeInSeconds}/>
            </div>
      )
}

export default Main