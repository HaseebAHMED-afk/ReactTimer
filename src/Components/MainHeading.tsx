import React from 'react'
import "../App.css"

const MainHeading: React.FC = (): JSX.Element => {
    return (
        <div>
            <h1 className="main-heading">Timer App</h1>
        </div>
    )
}

export default MainHeading