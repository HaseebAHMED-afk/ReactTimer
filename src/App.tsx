import React from 'react'
import './App.css';
import Main from './Components/Main/Main';
import MainHeading from './Components/MainHeading';

function App() {
  return (
    <div className="App">
     <MainHeading />
     <Main />
    </div>
  );
}

export default App;