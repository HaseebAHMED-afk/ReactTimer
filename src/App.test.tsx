import React from 'react';
import {shallow} from 'enzyme';
import App from './App';
import MainHeading from './Components/MainHeading';
import Main from './Components/Main/Main'

describe('App component renders', () => {
  let container:any;

  beforeEach(() => (container = shallow(<App />)))

  it('should render App components', ()=>{
    expect(container).toMatchSnapshot();
  })

  it('should render a div' , () =>{
    expect(container.find('div').length).toEqual(1)
  } )

  it('should render heading' , () => {
    expect(container.containsMatchingElement(<MainHeading />)).toEqual(true)
  })

  it('should render main component', () => {
    expect(container.containsMatchingElement(<Main />)).toEqual(true)
  })
})