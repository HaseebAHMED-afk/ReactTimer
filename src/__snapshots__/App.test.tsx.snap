// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders should render App components 1`] = `
<div
  className="App"
>
  <MainHeading />
  <Main />
</div>
`;
